src/utilities/photo_utils.tsx

```diff
@@ -21,13 +22,32 @@ import Metadata from '../types/metadata.type';
  *  timestamp
  * }
  */
+
+
+
+
+// Geolocation data from Web browser
+function getPosition(options?: PositionOptions): Promise<Position> {
+  return new Promise((resolve, reject) => {
+    navigator.geolocation.getCurrentPosition(resolve, reject, options)
+    }
+
+  );
+}
+
+
+
 export async function getPhotoMetadata(photo: Blob): Promise<Attachment["metadata"]> {
```

Comment:
I suggest having this function do just one thing – extract metadata from a photo. We might want to rename it `getMetadataFromPhoto`.
Put the new logic inside a new function (we could reuse the old name for this one):
```
export async function getPhotoMetadata(photo: Blob): Promise<Attachment["metadata"]> {
    // Attempt to extract timestamp and geolocation from the photo
    const metadataFromPhoto = await getMetadataFromPhoto(photo)

    // Set the metadata according to whether the timestamp and/or geolocation were set in the photo
    let photoMetadata = metadataFromPhoto
    const {timestamp: photoTimestamp, geolocation: photoGelolocation} = metadataFromPhoto
    if (photoTimestamp && 
        !(photoGelolocation.latitude && photoGelolocation.longitude) {
        const hrsDiff = difference between current time and metadataFromPhoto.timestamp in hours
        if (hrsDiff < 2) {
            photoMetadata = {timestamp: photoTimestamp, geolocation: await getCurrentGeolocation()}
        } 
    } else if (!(photoGelolocation.latitude && photoGelolocation.longitude) {
        photoMetadata = {timestamp: current time, geolocation: await getCurrentGeolocation()}
    }

    return photoMetadata
}
```

Since I've reused the name for this function, you won't need to change where old function got called in store.tsx.